packages/headless/src/features/search/search-actions-thunk-processor.ts

```diff
@@ -172,6 +172,7 @@ export class AsyncSearchThunkProcessor<RejectionType> {
     }
     if (
       state.didYouMean?.enableDidYouMean === false ||
+      state.didYouMean?.automaticallyCorrectQuery === false ||
```

Comment:
This multi line boolean check is getting hard to reason about.

I would suggest breaking it up with multiple private module functions that break down the logic very verbosely.
Long, verbose, descriptive is good, in this case.

Suggestion: 
```typescript
if(!state.didYouMean) {
   return null
}

const {enableDidYouMean, automaticallyCorrectQuery} = state.didYouMean;
const {results, queryCorrections} = successResponse;

if(!enableDidYouMean) {
   return null;
}

const shouldBeAutomaticallyRetried = results.length === 0 && queryCorrections.length !==  && automaticallyCorrectQuery;

if(!shouldBeAutomaticallyRetried) {
  return null;
}

// [...]
```